src/mapper.rs

```diff
@@ -61,6 +63,12 @@ impl<'m> Iterator for RemappedFrameIter<'m> {
     }
 }
 
+fn extract_class_name(full_path: &str) -> Option<&str> {
+    let after_last_period = full_path.split('.').last()?;
+    // If the class is an inner class, we need to extract the outer class name
+    after_last_period.split('$').next()
+}
```

Comment:
I'd like to add `.java` extension to this (https://github.com/getsentry/rust-proguard/pull/36#discussion_r1548171467), but it's `&str`. Do you have a suggestion how it can be achieved?

Reply:
It might be possible to achieve that using `Cow<'a, str>`. That type is either a reference (`&'a str`), or an owned `String` that you construct, for example using `format!("{classname}.java")`

Although you would have to change the type of `StackFrame::file` in that case, and convert all existing uses using `.into()` to convert the existing `&str` types into a `Cow`.

Reply:
Thanks, TIL about `Cow`.

I followed this tip and prepared a change, so we can add `.java` suffix (see here: https://github.com/wzieba/rust-proguard/pull/1).

But as I did it, I realized this might be not the best idea - I think I'd favor more not adding any extension file (which is not problematic for Sentry dashboard) rather than faking/assuming it, as it might be confusing. So I'd go with @romtsn suggestion (https://github.com/getsentry/rust-proguard/pull/36#discussion_r1549580502) to skip extension file altogether, as it is now.